R/as_prediction_surv.R

```diff
@@ -0,0 +1,59 @@
+#' @title Convert to a Survival Prediction
+#'
+#' @description
+#' Convert object to a [PredictionSurvival].
+#'
+#' @inheritParams mlr3::as_prediction
+#'
+#' @return [PredictionSurv].
+#' @export
+#' @examples
+#' library(mlr3)
+#' task = tsk("rats")
+#' learner = mlr_learners$get("surv.coxph")
+#' learner$train(task)
+#' p = learner$predict(task)
+#'
+#' # convert to a data.table
+#' tab = as.data.table(p)
+#'
+#' # convert back to a Prediction
+#' as_prediction_surv(tab)
+#'
+#' # split data.table into a list of data.tables based
+#' # on their survival times (ignoring censoring)
+#' tabs = split(tab, cut(tab$time, 3))
+#'
+#' # convert back to list of predictions
+#' preds = lapply(tabs, as_prediction_surv)
+#'
+#' # calculate performance in each group
+#' sapply(preds, function(p) p$score())
+as_prediction_surv = function(x, ...) {
+  UseMethod("as_prediction_surv")
+}
+
+
+#' @rdname as_prediction_surv
+#' @export
+as_prediction_surv.PredictionSurv = function(x, ...) { # nolint
+  x
+}
+
+
+#' @rdname as_prediction_surv
+#' @export
+as_prediction_surv.data.frame = function(x, ...) { # nolint
+  mandatory = c("row_ids", "time", "status")
+  optional = c("crank", "lp", "distr")
+  assert_names(names(x), must.include = mandatory)
+  assert_names(names(x), subset.of = c(mandatory, optional))
+
+  # FIXME: distr removed here, unsure how to combine them for PredictionSurv
+  # do.call(c, distr) is incredibly slow
```

Comment:
this is a known bottleneck, it affects all of proba; it's in parmetersetcollection (same bottleneck in paradox). doing my best to fix asap. not sure if best to uncomment and have it slow for now or just not have the method until fixed

Reply:
TBH, I thought I was doing something wrong. Better have it in there and slow than not supporting it.

Reply:
Okay can you uncomment then and include distr? Then I'll merge in. Bottleneck will be fixed at the end of the month once new package on cran